src/d3a/d3a_core/live_events.py

```diff
@@ -62,6 +64,9 @@ def apply(self, area):
             return False
 
         area.children = [c for c in area.children if c.uuid != self.area_uuid]
+        if len(area.children) == 0:
+            area._markets = AreaMarkets(area.log)
```

Comment:
Are you sure it wouldn't mess up with existing markets?
As far as I see `AreaMarkets ` contains current market too that are needed for ledgering the results

Reply:
You are right in general. However, in the UI case we dont keep the past markets, `AreaMarkets._market_rotation` will take care of deleting them anyway. This will "only" be a problem if you simulate with multiple future markets. But then again, this here, is only the case, when you delete the last child of an area. After deletion, there is no point of having markets anyway. So, IMO this reinitiating of the  `AreaMarkets` is unproblematic for this special case. Do you agree?

Reply:
Thanks for the explanation. Still in case of multiple future markets and  only one of the child being removed would distort the result. 

Nevertheless, you may merge this PR after dropping a comment in code to think of a better solution in case of multiple future spot markets.

Reply:
Thanks! Will leave a comment here and merge.